tasks/libs/common/utils.py

```diff
@@ -254,6 +254,7 @@ def get_build_flags(
         env['CGO_LDFLAGS'] = os.environ.get('CGO_LDFLAGS', '') + ' -Wl,--allow-multiple-definition'
 
     extra_cgo_flags = " -Werror -Wno-deprecated-declarations"
+    extra_cgo_flags += " -D__CGO__"  # Some C files include definitions that can cause duplicate symbols if included in CGo code (bpf_metadata.h) mainly
```

Comment:
IMHO this is a smell that the `bpf_metadata.h` file is getting included in places we don't want it. There are not that many top-level `.c` files (should be all the ones with `license` sections), so maybe we just include it in each place and add a build-time check to ensure it cannot be forgotten?